src/main/java/ru/skypro/homework/service/impl/UsersServiceImpl.java

```diff
@@ -0,0 +1,64 @@
+package ru.skypro.homework.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import ru.skypro.homework.dto.UpdateUser;
+import ru.skypro.homework.dto.User;
+import ru.skypro.homework.entity.UserEntity;
+import ru.skypro.homework.repository.UserRepository;
+import ru.skypro.homework.service.UsersService;
+import ru.skypro.homework.transformer.UserTransformer;
+
+import javax.transaction.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class UsersServiceImpl implements UsersService {
+
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
+    private final MyDatabaseUserDetailsService myDatabaseUserDetailsService;
```

Comment:
```suggestion
    private final UserDetailsService myDatabaseUserDetailsService;
```
инъекцию надо делать всегда по интерфейсу, а не реализации